src/BoundaryConditions/flux_calculations.jl

```diff
@@ -1,61 +1,130 @@
     given variable exchange coefficients. (e.g. for energy, or tracers)
 """
-function get_boundary_flux(
-    model,
-    bc::BulkFormulaCondition,
-    ρc::Fields.Field,
-    Y,
-    Ya,
-)
+
+function get_boundary_flux(model, bc::BulkFormulaDryTotalEnergy, ρc::Fields.Field, Y, Ya)
     FT = eltype(Y.base)
-    coefficients =
-        eltype(bc.coefficients) == FT ? bc.coefficients : bc.coefficients(Y, Ya)
-    c_sfc = bc.surface_field
 
-    ρ_1 = first_interior(Y.base.ρ)
-    ρc_1 = first_interior(ρc)
-    uv_1 = first_interior(Y.base.uv)
-    u_wind = norm(uv_1)
+    sfc = bc.surface_fields
 
-    flux =
-        Geometry.WVector(coefficients.Ch * u_wind * ρ_1 * (ρc_1 / ρ_1 - c_sfc))
+    # get relevant states of the first atmos internal point and the surface 
+    p = calculate_pressure(
+        Y,
+        Ya,
+        model.base,
+        model.thermodynamics,
+        model.moisture,
+        model.parameters,
+        FT,
+    )
+
+    qt_sfc = FT(0)
+    cv_d = TD.cv_m(model.parameters, FT)
+
+    # internal energies
+    e_int_in = Fields.level(TD.air_temperature_from_ideal_gas_law.(model.parameters, p, Y.base.ρ) .* cv_d, 1)
+    e_int_sfc = sfc.surface_temperature .* cv_d
+
+    # horizontal velocity
+    uh = Geometry.UVWVector.(Y.base.uh)
+    u_in = Fields.level(uh.components.data.:1, 1)
+    v_in = Fields.level(uh.components.data.:2, 1)
+    u_sfc = sfc.surface_horizontal_velocity[1]
+    v_sfc = sfc.surface_horizontal_velocity[2]
+
+    # height
+    z_in = similar(e_int_in)
+    parent(z_in) .= parent(Fields.coordinate_field(axes(e_int_in)).z)[1]
+    z_sfc = sfc.surface_height
+
+    # density
+    ρ_in = Fields.level(Y.base.ρ,1)
+    ρ_sfc = Fields.level(Y.base.ρ,1)
+
+    # broadcast SurfaceFluxes.jl functions
+    bulk_flux = map_fluxes.(parent(e_int_in), e_int_sfc, parent(u_in), u_sfc, parent(v_in), v_sfc, parent(z_in), z_sfc, parent(ρ_in), parent(ρ_sfc), qt_sfc; model = model, bc = bc)
+
+    flux = Geometry.Contravariant3Vector.(parent(bulk_flux)[1]) # TODO: change when ClimaCore enables Fields for specifying boundary conditions (#325)
+end
+
+function map_fluxes(e_int_in, e_int_sfc, u_in, u_sfc,  v_in, v_sfc, z_in, z_sfc, ρ_in, ρ_sfc, qt_sfc; model = nothing, bc = nothing)
+
+    FT = eltype(e_int_in)
+
+    # recombine into a horizontal velocity vector
+    uh_in = [u_in, v_in]
+    uh_sfc = [u_sfc, v_sfc]
+
+    # thermodynamic states
+    ts_in = TD.PhaseEquil_ρeq(model.parameters, ρ_in, e_int_in, qt_sfc) 
+    ts_sfc = TD.PhaseEquil_ρeq(model.parameters, ρ_sfc, e_int_sfc, qt_sfc)  
+
+    # collect states
+    state_sfc = SF.SurfaceValues(z_sfc, uh_sfc, ts_sfc) 
+    state_in = SF.InteriorValues(z_in, uh_in, ts_in)
+
+    # collect all surface conditions
+    kwargs = (; state_in, state_sfc)
+    sc = SF.Coefficients{FT}.(; kwargs..., Cd = bc.coefficients.Cd, Ch = bc.coefficients.Ch, z0m = FT(0), z0b = FT(0))
+
+    uf = UF.Businger()
+    bulk_flux = SF.sensible_heat_flux(model.parameters, sc.Ch, sc, uf) 
```

Comment:
The idea behind SF.jl is not to use it's internal functions such as `SF.sensible_heat_flux`,  but to call the user facing function `surface_condition`, i.e.:

`result = SF.surface_conditions(param_set, sc, universal_func, scheme)`

and get the fluxes from:

`bulk_flux = results.shf`

and also:
`bulk_lhf_flux = results.lhf`


There are many reason to do that
1. the interface will not change with changing in SF
2. you get other properties (obukhuv length, similarity profiles and such at the same time)
3. the user does need to "know" the theory to used it